internal/reconciliations/istio/installation_test.go

```diff
@@ -1115,11 +1115,64 @@ var _ = Describe("Installation reconciliation", func() {
 		Expect(mockClient.uninstallCalled).To(BeFalse())
 
 		Expect(istioCR.Status.Conditions).ToNot(BeNil())
-		Expect((*istioCR.Status.Conditions)).To(HaveLen(1))
+		Expect(*istioCR.Status.Conditions).To(HaveLen(1))
 		Expect((*istioCR.Status.Conditions)[0].Type).To(Equal(string(operatorv1alpha1.ConditionTypeReady)))
 		Expect((*istioCR.Status.Conditions)[0].Reason).To(Equal(string(operatorv1alpha1.ConditionReasonIstioCRsDangling)))
 		Expect((*istioCR.Status.Conditions)[0].Status).To(Equal(metav1.ConditionFalse))
 	})
+
+	It("Should have all istio components labeled with kyma-project.io/module=istio label", func() {
+		numTrustedProxies := 1
+		istioCR := operatorv1alpha1.Istio{ObjectMeta: metav1.ObjectMeta{
+			Name:            "default",
+			ResourceVersion: "1",
+			Annotations:     map[string]string{},
+		},
+			Spec: operatorv1alpha1.IstioSpec{
+				Config: operatorv1alpha1.Config{
+					NumTrustedProxies: &numTrustedProxies,
+				},
+			},
+		}
+		istiod := createPod("istiod", gatherer.IstioNamespace, "discovery", istioVersion)
```

Comment:
why this is not checked `istiod` pod you add for this test for the label after `reconcile()` ?

could you add all resources from:
https://github.com/kyma-project/istio/blob/bc74d68e1288995a0be9f8d2fbc95f195721ff94/internal/reconciliations/istio/metadata.go#L24-L49

we do not have integration test, at lest fully test everything in unit test would be great (regression proved)

Reply:
I'm testing all 3 cases in a function based on the type. As long as the type of resource is in the list and contains a `operator.istio.io/component` label which indicates that a resource is part of istio installation, it will be labeled with a module. I don't see a need for integration test, as unit for this function covers all needed statements, so the function should work.